packetbeat/protos/protos.go

```diff
@@ -231,11 +231,17 @@ func (protocols ProtocolsStruct) BpfFilter(with_vlans bool, with_icmp bool) stri
 
 func (protos ProtocolsStruct) Register(proto Protocol, plugin ProtocolPlugin) {
 	protos.all[proto] = plugin
+	var success bool = false
```

Comment:
This logic would be well suited for a [type switch](https://golang.org/doc/effective_go.html#type_switch) where the default case logs the warning.